crates/sqlbuiltins/src/validation.rs

```diff
@@ -165,3 +168,13 @@ pub fn validate_copyto_dest_format_support(dest: &str, format: &str) -> Result<(
         })
     }
 }
+
+pub fn validate_copyto_format_partition_support(format: &str) -> Result<()> {
+    if matches!(format, |"parquet"| "json") {
```

Comment:
I feel like we could have a method on the enum or something that could do this validation rather than on a string. Is this extension? I worry about this falling out of sync with other variations of the enum?